Base on official PHP 7.0 Alpine image

kusmierz opened the pull request:
see #16 and #20

kusmierz committed:
#16 php 7.0 support

mikehaertl commented on Dockerfile:
```diff
@@ -1,35 +1,37 @@
- && ln -s ../../upload_large_dumps.ini /etc/php5/cli/conf.d
+## Add Tini
+RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ tini
+ENTRYPOINT ["/usr/bin/tini", "--"]
```
@kusmierz Why do you add a init system? Why not rely on the default `CMD` from the official php-7 image?

mikehaertl replied:
Never mind, i found the related discussion and the reference to #17

kusmierz committed:
Little bit fixes (https and Tini is now higher, so it should be faster rebuild, when new adminer version will arrive).

mathroc commented on Dockerfile:
```diff
@@ -16,21 +16,21 @@ RUN set -x \
 
 RUN mkdir -p /var/www
-ADD https://www.adminer.org/static/download/$ADMINER_VERSION/adminer-$ADMINER_VERSION.php /var/www/index.php
+ADD http://www.adminer.org/static/download/$ADMINER_VERSION/adminer-$ADMINER_VERSION.php /var/www/index.php
```
why the change to https ?

kusmierz replied:
By mistake, I've reverted it in e6f58f21009ee1d977982f4e4e59df1a6a708c13

mathroc commented:
seems good to me! :+1:

kusmierz commented:
I just thinking about difference between:

```dockerfile
ADD http://www.adminer.org/static/download/$ADMINER_VERSION/adminer-$ADMINER_VERSION.php /var/www/index.php
```

and downloading it using `wget` (system one or from apk, but system's is without https support).

Mainly first one will download each and every time on build, and wget will probably use cached version. @mathroc what do you think?

kusmierz commented on php.ini:
```diff
@@ -0,0 +1,4 @@
+upload_max_filesize = 2000M
+post_max_size = 2000M
+memory_limit = -1
```
Is it good idea to set here unlimited execution time and no memory limit?

mathroc replied:
exports & imports can take a very long time, not sure what would be enough and reasonnable

clue replied:
:+1: Afaict these have only been moved (see also #8 and #4).

mathroc commented:
@kusmierz if `ADD` works I'm fine with it. (dont' mind either way)

kusmierz committed:
Update Dockerfile

Always download latest version of adminer.

mikehaertl commented on Dockerfile:
```diff
@@ -1,35 +1,35 @@
-EXPOSE 80
+## install adminer and default theme
+RUN mkdir -p /var/www
+ADD http://www.adminer.org/latest.php /var/www/index.php
```
We should create tags for this image, that are in sync with adminer (e.g. `clue/adminer:4.2.4`). So instead of `latest` we should point to the specific version `https://www.adminer.org/static/download/4.2.4/adminer-4.2.4.php`

kusmierz replied:
I think so too, but as long as we have no tags here, I think it would be better to have latest.

Anyway, we could have tags like apache, alpine-php56, alpine-php7, etc. But do you think is worth for?

mikehaertl replied:
I don't think we need different variants (alpine, php56, ...). But I do think, we should tag the releases here, starting now. That's why I'm still for changing the URL.

kusmierz replied:
@clue what do you think?

clue replied:
> But I do think, we should tag the releases here, starting now. 

See also #6 and #5. I think this is a bad for for this PR, but I'm open to discussing this. So I'd vote for removing this from this PR for now in order to ease the discussion :+1:

clue commented on Dockerfile:
```diff
@@ -1,35 +1,35 @@
 
-ADD supervisord.conf /etc/supervisor/conf.d/supervisord.conf
-CMD supervisord -c /etc/supervisor/conf.d/supervisord.conf
+CMD php -S 0.0.0.0:80 -t /var/www/
```
Do we really want to use PHP's built-in webserver here? (open for discussion)

kusmierz replied:
I thought about this for some time. @mathroc proposed this solution in #20.

I even made `Dockerfile` with nginx and php-fpm before, but I think @mathroc is right (but only in this particular case, see good explanation in https://github.com/smebberson/docker-alpine/tree/master/alpine-nginx, "Aren't you only supposed to run one process per container?").

But, as we think about adminer - is dev tool, obviously. When we add nginx (at least 2 processes) and php-fpm (at least another 2) + supervisor or even smaller [s6](http://www.skarnet.org/software/s6/) instead only one (php's built-in webserver) it becomes a little bit overcomplicated.

+1 for @mathroc solution here.

clue replied:
I think you're making some very valid points here :+1:

However, I'm not sure this should be discussed as part of this PR. I understand where you're coming from, but this particular statement is very well worth discussing IMHO:
> But, as we think about adminer - is dev tool, obviously

Would you care to file a separate issue for this? :+1: 
As an alternative, we *could* also discuss this in a separate PR, as we can also change to this in our current image without switching to Alpine first.

clue replied:
ping, any thoughts on this, anybody?

IMO this change alone already warrants a separate PR / discussion.

kusmierz replied:
There was PR/issue about it, created by @mathroc #20.

mathroc commented:
@clue about your [comment on versionning](https://github.com/clue/docker-adminer/pull/19#discussion_r59180165], with the docker hub we can provide both the latest and tagged release. in other images I'm doing it this way:

 * master branch have SOMETHING_VERSION=latest adn configured on docker hub to auto build as tag `latest`
 * releases branch with fixed version and tags `Vx.x.x` and tags are set to build automatically on docker hub too

anyway, this can be done in a second time, @kusmierz what do you think about sticking with latest for the time being and working on tags later ?

kusmierz commented:
:+1:

kusmierz commented:
@mathroc any chances to push this version into the wild?

mathroc commented:
I don't have more comment for this PR. it's ready as far as I'm concerned. it's up to @clue to either merge this into master or in another branch (to tag it `alpine` un docker hub until it's a bit more tested)

mikehaertl commented:
@mathroc @kusmierz I think it's not helpful for @clue to have 3 open PRs here now. Could you maybe close the irrelevant ones? Would be great to finally see some progress here.

kusmierz committed:
Smaller image

kusmierz commented on Dockerfile:
```diff
@@ -1,35 +1,37 @@
+ADD php.ini /usr/local/etc/php/conf.d/php.ini
+
+## install adminer and default theme
+ENV ADMINER_VERSION 4.2.4
```
Is it better download "latest" or having const with version?

mikehaertl replied:
@kusmierz See the discussion here: https://github.com/clue/docker-adminer/issues/5

kusmierz replied:
Great, thanks @mikehaertl I knew I've seen it somewhere :)

clue replied:
There is some discussion about this in #5, however until we get to solve this I would vote to not introduce any additional changes in this PR :+1: (In other words, revert this change and)

After all, this PR ought to be about Alpine, right? :)

kusmierz replied:
Done.

kusmierz committed:
Tini has been relocated to /sbin/tini.

Tini has been relocated to /sbin/tini.
Please update your scripts to use /sbin/tini going forward.
/usr/bin/tini has been preserved for backwards compatibility in Alpine 3.4, but WILL BE REMOVED in Alpine 3.5.

mathroc commented:
@clue do you think it can be merged once rebased (/ping @kusmierz ) ? or do you have other suggestions ?

clue commented:
> @clue do you think it can be merged once rebased

Afaict there are still some outstanding, uncommented issues in this PR, other than that I'm good with this change :+1:

kusmierz committed:
#21 Downloading "latest" adminer instead tagged one (see #5)

kusmierz committed:
Merge branch 'feature/16-alpine-php70' of github.com:kusmierz/docker-adminer into feature/16-alpine-php70

mathroc commented on Dockerfile:
```diff
@@ -1,35 +1,35 @@
-FROM ubuntu-debootstrap:14.04
+FROM php:7.0.7-alpine
```
maybe `7-alpine` or at least `7.0-alpine` ?

kusmierz replied:
`7-alpine` is dangerous - why if they change something in ie. 7.2?

mathroc replied:
correct

mathroc commented on Dockerfile:
```diff
@@ -1,35 +1,35 @@
-  php5-pgsql php5-mysql php5-sqlite php5-mssql \
-  wget
+## Install php and dependencies
+RUN set -ex \
```
is `set -ex` really useful ?

kusmierz replied:
and why not?

mathroc replied:
well, at least the `-e` part does not seems very useful as each commands are joined with `&&`. I'm not sure about the `'x` part. it's used to print the execution trace, that's it ?
tbh, it's just for curiosity as I don't think it has any impact on the image in the end

mathroc commented on Dockerfile:
```diff
@@ -1,35 +1,35 @@
-WORKDIR /var/www
-RUN chown www-data:www-data -R /var/www
+## Add Tini
+RUN apk add --no-cache --repository http://dl-cdn.alpinelinux.org/alpine/edge/community/ tini
```
alpine version of official php images has been upgraded and it includes tini in the defauls packages list now. no need for the custom repository anymore (might even not work anymore)

kusmierz replied:
Yeah, I saw it before, but had no time to update it. Thanks.

mathroc commented:
@kusmierz I've made a few more comments, do you mind taking a look and rebasing the PR ?

@clue is there still issues you'd like to be taken care for this PR? moving to `php -S` in another PR before this one maybe ?

kusmierz committed:
Small fixes (@mathroc comments)

kusmierz committed:
#15 #16 ODBC dirty fix

kusmierz commented:
@clue do we really need ODBC (#15)? It makes only problems (749192d)...
